openeo/extra/job_management.py

```diff
@@ -822,3 +825,104 @@ def persist(self, df: pd.DataFrame):
         self._merge_into_df(df)
         self.path.parent.mkdir(parents=True, exist_ok=True)
         self.df.to_parquet(self.path, index=False)
+
+
+class UDPJobFactory:
+    """
+    Batch job factory based on a parameterized process definition
+    (e.g a user-defined process (UDP) or a remote process definitions),
+    to be used together with :py:class:`MultiBackendJobManager`.
+    """
+
+    def __init__(
+        self, process_id: str, *, namespace: Union[str, None] = None, parameter_defaults: Optional[dict] = None
+    ):
+        self._process_id = process_id
+        self._namespace = namespace
+        self._parameter_defaults = parameter_defaults or {}
+
+    def _get_process_definition(self, connection: Connection) -> dict:
+        if isinstance(self._namespace, str) and re.match("https?://", self._namespace):
+            return self._get_remote_process_definition()
+        elif self._namespace is None:
+            return connection.user_defined_process(self._process_id).describe()
+        else:
+            raise NotImplementedError(
+                f"Unsupported process definition source udp_id={self._process_id!r} namespace={self._namespace!r}"
+            )
+
+    @functools.lru_cache()
+    def _get_remote_process_definition(self) -> dict:
```

Comment:
this "remote process definition" handling can be refactored to a separate utility I think, does not have to be part of UDFJobFactory